RIOT-OS/RIOT
# tests/periph_uart: Add uart robot tests

### Contribution description
This adds some robot framework based tests that are compatible with the HiL testing.  It is off the robot i2c pr and ported from @yegorich [uart test](https://github.com/yegorich/RIOT/commit/8e67d0165bf05a6be11acdacfb60896383b26e4f)

### Testing procedure
Either connect up PHiLIP and a DUT and run:
`BOARD=<your favorite board> PHILIP_PORT=<your favorite PHiLIP port> make -C tests/periph_uart/ flash robot-test`<\br>
or wait for the nightlies and check the HiL results

### Issues/PRs references
depends on #10147
